src/util/simplify_expr.cpp

```diff
@@ -165,8 +165,12 @@ bool simplify_exprt::simplify_function_application(exprt &expr)
 {
   const function_application_exprt &function_app =
     to_function_application_expr(expr);
-  irep_idt func_id =
-    function_app.function().find(ID_expression).get(ID_identifier);
+
+  if(function_app.function().id() != ID_symbol)
+    return true;
+
+  const irep_idt func_id =
```

Comment:
const ref?

Reply:
Deliberately didn't -- a problem in the design of the simplifier is that it takes a non-const ref to an expression, and then changes that expression (say as opposed to building a new expression and then returning it). This may well invalidate those const refs taken at the beginning. With this design, it's safer to copy.